content/references/rippled-api/api-conventions/request-formatting.md

```diff
@@ -1,70 +1,133 @@
+
+Always include a `Content-Type` header with the value `application/json`.
+
+If you plan on making multiple requests, use [Keep-Alives](http://tools.ietf.org/html/rfc7230#section-6.3) so that you do not have to close and re-open the connection in between requests.
+
+Send request body as a [JSON](https://en.wikipedia.org/wiki/JSON) object with the following fields:
+
+
+| Field               | Type      | Description                                |
+|:--------------------|:----------|:-------------------------------------------|
+| `method`            | String    | The name of the [API method](public-rippled-methods.html). |
+| `params`            | Array     | _(Optional)_ A **one-item array** containing a nested JSON object with the parameters to this method. You may omit this field if the method does not require any parameters. |
+
+The object inside the `params` array can contain the following fields:
+
+| Field               | Type      | Description                                |
+|:--------------------|:----------|:-------------------------------------------|
+| `api_version`       | Number    | _(Optional)_ The API version to use. If omitted, use version 1. For details, see [API Versioning](#api-versioning). [New in: rippled 1.5.0][] |
+| (Method Parameters) | (Various) | Provide any parameters to the method here. |
+
+See [Response Formatting](response-formatting.html) for the response from the server.
+
+## Commandline Format
+
+Put the API method name after any normal (dash-prefaced) commandline options, followed by a limited set of parameters, separated by spaces. For any parameter values that might contain spaces or other unusual characters, use single-quotes to encapsulate them. Not all methods have commandline API syntax. For more information, see [Commandline Usage](https://xrpl.org/commandline-usage.html#client-mode-options).
+
+The commandline calls JSON-RPC, so its responses always match the JSON-RPC [response format](response-formatting.html).
+
+The commandline always uses the latest [API version](#api-versioning).
+
+**Caution:** The commandline interface is intended for administrative purposes only and is _not a supported API_. New versions of `rippled` may introduce breaking changes to the commandline API without warning!
+
+## API Versioning
+
+The `rippled` server uses a single integer to identify the API version to use. The first API version is `1`; currently, this is the only version of the `rippled` API. (There is no API version 0.) [New in: rippled 1.5.0][]
+
+Future versions of `rippled` that introduce breaking changes will introduce a new API version number, such as `2`. The server will support a range of API versions, which it reports in the `version` API method. <!-- TODO: add a link when `version` method is documented. -->
+
```

Comment:
I'll add a note about this, but leave it commented out until `rippled` supports multiple API versions, since it doesn't make sense to say you can use different API versions per request but the only API version you can use is version 1.

Reply:
Great! Please also consider:
-- We will support a range of consecutive API versions in rippled. 
-- The "api_version" field in the request is optional, and default to version 1. When version 1 is no longer in the supported version range, however, a request without the "api_version" field will fail with error "invalid_API_version". I.e. if we stop supporting version 1 API, "api_version" becomes a required field.